### DISCUSSION TITLE
Firstpass height and width with hi-res fix in img2img?

### DISCUSSION THREAD

**Discussion by Drakmour** [zero01101/openOutpaint#136] [Q&A] [answered]
I can't find where to put the initial resolution for hi-res fix in img2img of OpenOutpaint. Like firstpass height and width in text2img. Is there such option? I've read that right now it decrease the resolution x2 and then generate, but if my image is more than 1024 and is not square it still result in morbid multiple characters. I'd like to have an option to make firstpass 512x512 and then resize to my end resolution. It 99% works in text2img and should work in img2img. In text2img I get random result, while in img2img I can draw a sketch of pose of character myself to have much more control of the result.

**Comment by zero01101** [comment, marked as answer]
hey, the auto txt2img hrfix option does indeed only work on dream tool usage since it's strictly a txt2img function in A1111 - the inpaint full resolution option would be kind of the "closest" related option for img2img.  in any case, the hr fix values are "magical" and automated; the firstpass values should just be half of the reticle's size; i.e. set your reticle to 1024x1024 and check the box under stable diffusion settings, and firstphase width/height will be 512px

[edit]  ah, i missed the not square detail; yeah, the firstpass w/h values may need to be revisited for custom reticle sizes

**Reply by zero01101** [reply]
yep, non-square drawn boundaries were using the resolution slider value for auto hrfix x/y values, and now uses the actual drawn boundary box instead. thanks for finding that 👍 

@Drakmour if you checkout the testing branch via `git checkout testing` could you test if it's working more along your expectations?

**Reply by Drakmour** [reply]
Thank you so much for a quick response! :-) I'm not very familiar with git mechanics, I usually just install all soft without git commands. Ho to  "git checkout testing"? I need to put it in users bat file as --xformers and other stuff?

**Reply by zero01101** [reply]
ah, no; that'd be just run in your command prompt or terminal from within the directory you installed openOutpaint into; i.e. assuming windows, if you installed it into c:\oo\ then in command prompt you'd do something like 
```
cd \oo
git checkout testing
```

**Comment by zero01101** [comment]
converting to an issue as yeah that's a bug - https://github.com/zero01101/openOutpaint/issues/137
